system/logging/Logger.cfc

```diff
@@ -456,4 +474,46 @@ component accessors="true" {
 		return canLog( this.logLevels.DEBUG );
 	}
 
+	private boolean function checkForComplexObjects( required any value ){
+		if ( isNull( arguments.value ) ) {
+			return false;
+		}
+
+		if ( isSimpleValue( arguments.value ) ) {
+			return false;
+		}
+
+		// If the object has a $toString() method, then we consider it safe
+		if ( isObject( arguments.value ) AND structKeyExists( arguments.value, "$toString" ) ) {
+			return false;
+		}
+
+		// CFML Exceptions are safe
+		if ( isCFMLException( arguments.value ) ) {
+			return false;
+		}
+
+		if ( isArray( arguments.value ) ) {
```

Comment:
In all reality, if we are going to NOT allow anything complex, I would say don't even check for here down, why go through the trouble of iterate through each of the values, this is time consuming, error proned and just slow.